### ISSUE TITLE
Patching Framework7 for Turbolinks compatibility with Rails 4.1 variants?

### ISSUE THREAD

**Issue by reconstructions** [reed/turbolinks-compatibility#50]
I have been using Framework7 with Rails 4.1 variants and the UI is amazing.  I have had to use some work arounds to replace the Rails form and link helpers which don’t work with Framework7, but have gotten as far as having CRUD working iOS style.  It took me only one week to build what looks exactly like an iOS app with CRUD for 8 controllers.  The performance is incredible, I have fooled some testers into thinking it is a native app, even running on an iPhone 4S (i.e. old hardware).

The problem is that I can’t get my pages to automagically reload the changes the CRUD makes to the DB, and Framework7’s routing, caching, and reloading relies heavily on manual event handler calls.  It took me one week to build new templates for 8 controllers worth of CRUD, but I couldn’t get even one of them to reload properly after a couple days of work with Framework7's handlers.

In the [recipes I have been using](https://github.com/twss/framework7-rails),  Framework7 binds globals to the window object, so it is breaking Turbolinks. 

```
window.F7H = {
  app: new Framework7(),
  dom: Dom7
};

window.Phone = {
  Views: {}
};

Phone.Views.Main = F7H.app.addView('.view-main', {
  dynamicNavbar: true
});
```

 It looks to me like it might be fairly simple to patch this, but I am really a novice at this sort of thing.  I also [posted an issue](https://github.com/nolimits4web/Framework7/issues/406) on Framework7’s repository talking up Turbolinks and asking for their help.

**Comment by reconstructions** [comment]
I'm going to close this one because I finally got it working.  Something had broken my Turbolinks gem, and because it happened at the same time I configured Framework7 they seemed to be connected.  I bumped the version on Turbolinks and that fixed that problem.

I still had trouble getting Framework7 to play nice with Turbolinks, because both normally rely on XHR requests and both can be used to do pushState URL rewriting.  I got the best of both worlds by configuring Framework7 to switch off AJAX for most of its requests and caching like this:

```
window.MyApp = {
  app: new Framework7({
    cache: false,
    ajaxLinks: "foo",
  }),
  dom: Framework7.$
};
```

And then Turbolinks did the rest with automagical refreshing of the pages on CRUD success, pretty url rewriting, and so forth.  An amazing combo for Rails-based mock iOS implementations, particularly using variants.
